fetch.bs

```diff
@@ -2697,15 +2700,86 @@ functionality.
+ <dd>A <a for=/>request</a>.
+
+ <dt><dfn export for="deferred fetch record">inactivity delay</dfn> (default null)
+ <dd>Null or a <a>duration</a>.
+
+ <dt><dfn export for="deferred fetch record">invoke state</dfn> (default "<code>deferred</code>")
+ <dd>
+  <p>"<code>deferred</code>", "<code>scheduled</code>", "<code>terminated</code>",
+  "<code>aborted</code>", or "<code>activated</code>".
+
+  <p class=note>This value can be modified <a>in parallel</a>. There could be a race condition where
+  the <code>Document</code> object's <a for=/>event loop</a> might change it to
+  "<code>deferred</code>" at the same time that it is changed to "<code>activated</code>". UAs can
+  mitigate this race condition in an <a>implementation-defined</a> manner.
+</dl>
+
+
 <hr>
 
-<p>When a <a for=fetch>fetch group</a> is
-<dfn export for="fetch group" id=concept-fetch-group-terminate>terminated</dfn>, for each associated
-<a for="fetch group">fetch record</a> whose <a for="fetch group">fetch record</a>'s
-<a for="fetch record">controller</a> is non-null, and whose <a for="fetch record">request</a>'s
-<a>done flag</a> is unset or <a for=request>keepalive</a> is false,
-<a for="fetch controller">terminate</a> the <a for="fetch group">fetch record</a>'s
-<a for="fetch record">controller</a>.
+<p>When a <a for=fetch>fetch group</a> <var>fetchGroup</var> is
+<dfn export for="fetch group" id=concept-fetch-group-terminate>terminated</dfn>:
+
+<ol>
+ <li><p><a for=list>For each</a> <a for="fetch group">fetch record</a> <var>record</var> of
+ <var>fetchGroup</var>'s <a for="fetch group">fetch records</a>, if <var>record</var>'s
+ <a for="fetch record">controller</a> is non-null and <var>record</var>'s
+ <a for="fetch record">request</a>'s <a>done flag</a> is unset and <a for=request>keepalive</a> is
+ false, <a for="fetch controller">terminate</a> <var>record</var>'s
+ <a for="fetch record">controller</a>.
+
+ <li>
+  <p><a for=list>For each</a> <a for="fetch group">deferred fetch record</a>
+  <var>deferredRecord</var> of <var>fetchGroup</var>'s
+  <a for="fetch group">deferred fetch records</a>:
+
+  <ol>
+   <li><p>Set <var>deferredRecord</var>'s <a for="deferred fetch record">invoke state</a> to
+   "<code>terminated</code>".
+
+   <li><p><a for=/>Fetch</a> <var>deferredRecord</var>'s
+   <a for="deferred fetch record">request</a>.
+  </ol>
+ </li>
+</ol>
+
+<p>When a <a for=fetch>fetch group</a> <var>fetchGroup</var> is
+<dfn export for="fetch group" id=concept-fetch-group-deactivate>deactivated</dfn>:
+<a for=list>For each</a> <a>deferred fetch record</a> <var>deferredRecord</var> of
+<var>fetchGroup</var>'s <a for="fetch group">deferred fetch records</a> whose
+<a for="deferred fetch record">inactivity delay</a> is not null:
+
+<ol>
+ <li><p>Set <var>deferredRecord</var>'s <a for="deferred fetch record">invoke state</a> to
+ "<code>scheduled</code>".
+
+ <li>
+  <p>Run the following steps <a>in parallel</a>:
```

Comment:
```
Run the following steps in parallel
```

Could you please help me understand how the following steps are run `in parallel`? Step 1 says it has to wait, and Step 2 checks if the record is not `scheduled`, step 3 sets state to `activated`. How can these all be run in the same time?

Reply:
"Run these steps in parallel" means "Run these steps in a separate thread/process". The steps run in parallel with the rest of the execution, not with each other.